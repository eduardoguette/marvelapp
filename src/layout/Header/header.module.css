.header {
  background-color: var(--secondary);
  max-height: 5.35rem;
  padding: 1rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
}




@media (width > 640px) {
  .header {
    padding: 1rem 3rem 1rem 3rem;

  }
}

@media (width > 830px) {
  .container {
    padding: 16px 48px 16px 48px;
  }
}