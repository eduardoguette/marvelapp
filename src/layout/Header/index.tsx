import logo from '@/assets/logo.svg'
import { LikeCounter } from '@/components/Characters'
import { Link } from 'react-router-dom'
import styles from './header.module.css'
export const Header = () => {
  return (
    <header className={styles.header}>
      <Link to={'/'}>
        <img src={logo} alt="Logo Marvelapp" />
      </Link>
      <LikeCounter />
    </header>
  )
}
