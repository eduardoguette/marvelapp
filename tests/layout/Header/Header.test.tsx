import { render, screen } from '@testing-library/react'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { describe, expect, it, vi } from 'vitest'
 
import { Header } from '../../../src/layout/Header/index'

 
vi.mock('@/components/Characters', () => ({
  LikeCounter: () => <div data-testid="like-counter" />
}))

describe('Header Component', () => {
  it('debería renderizar correctamente', () => {
    const { container } = render(
      <Router>
        <Header />
      </Router>
    )
    expect(container).toMatchSnapshot()
  })

  it('debería contener un enlace a la ruta "/"', () => {
    render(
      <Router>
        <Header />
      </Router>
    )

    const linkElement = screen.getByRole('link')
    expect(linkElement.getAttribute('href')).toBe('/')
  })

  it('debería renderizar el logo con las propiedades adecuadas', () => {
    render(
      <Router>
        <Header />
      </Router>
    )

    const logoElement = screen.getByAltText('Logo Marvelapp')
    expect(logoElement).not.toBeNull()
    expect(logoElement.getAttribute('src')).toBe('/src/assets/logo.svg')
    expect(logoElement.getAttribute('width')).toBe('130')
    expect(logoElement.getAttribute('height')).toBe('52')
 
  })

  it('debería renderizar el componente LikeCounter', () => {
    render(
      <Router>
        <Header />
      </Router>
    )

    const likeCounterElement = screen.getByTestId('like-counter')
    expect(likeCounterElement).not.toBeNull()

  })
})
