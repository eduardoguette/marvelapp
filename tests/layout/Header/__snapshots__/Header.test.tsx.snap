// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > deberÃ­a renderizar correctamente 1`] = `
<div>
  <header
    class="_header_bdf1c4"
  >
    <a
      href="/"
    >
      <img
        alt="Logo Marvelapp"
        height="52"
        src="/src/assets/logo.svg"
        width="130"
      />
    </a>
    <div
      data-testid="like-counter"
    />
  </header>
</div>
`;
